### DISCUSSION TITLE
Add LVGL to RP2040 firmware

### DISCUSSION THREAD

**Discussion by Carlos-TGW** [micropython/micropython#15210] [RP2040 / Pico]
Can you add LVGL to RP2040 firmware?

**Comment by andrewleech** [comment]
There isn't any standard way to build this into rp2040 or any other port currently, but there are some active developments under way if you search for lvgl here you might find a few discussions. 
The main blocker is a need for a standard method of non-blocking / background IO for the various display drivers.

**Comment by rajsiyer** [comment]
Actually, there is sufficient guidance at[https://github.com/lvgl/lv_micropython](https://github.com/orgs/micropython/discussions/url) There is a 6-step process given for Raspberry pi Pico. I have followed the steps and built it using Cmake. The build results in a firmware.uf2 file that can be copied onto the Pico in the usual way.https://github.com/lvgl/lv_micropython
